weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts

```diff
@@ -313,40 +309,58 @@ const useCallsNoExpansion = (
         ? {include_total_storage_size: true}
         : null),
     };
-    const onSuccess = (res: traceServerTypes.TraceCallsQueryRes) => {
-      loadingRef.current = false;
-      setCallRes(res);
-    };
-    const onError = (e: any) => {
-      loadingRef.current = false;
-      console.error(e);
-      setCallRes({calls: []});
-    };
-    getTsClient().callsStreamQuery(req).then(onSuccess).catch(onError);
   }, [
-    opts?.skip,
-    opts?.includeCosts,
-    opts?.includeFeedback,
-    opts?.includeTotalStorageSize,
     entity,
     project,
-    deepFilter.opVersionRefs,
-    deepFilter.inputObjectVersionRefs,
-    deepFilter.outputObjectVersionRefs,
-    deepFilter.parentIds,
-    deepFilter.traceId,
-    deepFilter.callIds,
-    deepFilter.traceRootsOnly,
-    deepFilter.runIds,
-    deepFilter.userIds,
+    deepFilter,
     limit,
     offset,
     sortBy,
     query,
     columns,
-    getTsClient,
+    opts?.includeCosts,
+    opts?.includeFeedback,
+    opts?.includeTotalStorageSize,
   ]);
 
+  // Create a hash of the current request parameters
+  const currentRequestHash = useMemo(() => {
+    return JSON.stringify(makeRequest());
```

Comment:
I'm guessing `JSON.stringify` sorts keys?

Reply:
you are generating a new object each time, try:
```javascript
const payload = {"a": 1, "b": 2}
JSON.stringify(payload)
JSON.stringify(payload)
```

Reply:
oh, but because I DRYed it out with the `makeRequest` function we are actually assigning twice. good catch

Reply:
Hmm, well because its in the `useMemo` hook it should only ever fire when one of the makeRequest dependencies changes: 
```
entity,
    project,
    deepFilter,
    limit,
    offset,
    sortBy,
    query,
    columns,
    opts?.includeCosts,
    opts?.includeFeedback,
    opts?.includeTotalStorageSize,
```